core/protocols/protocol-core/src/main/java/software/amazon/awssdk/protocols/core/StringToInstant.java

```diff
@@ -50,21 +51,34 @@ public Instant convert(String value, SdkField<Instant> field) {
             case ISO_8601:
                 return DateUtils.parseIso8601Date(value);
             case UNIX_TIMESTAMP:
-                try {
-                    return DateUtils.parseUnixTimestampInstant(value);
-                } catch (NumberFormatException e) {
-                    throw SdkClientException.builder()
-                                            .message("Unable to parse date : " + value)
-                                            .cause(e)
-                                            .build();
-                }
+                return safeParseDate(DateUtils::parseUnixTimestampInstant).apply(value);
+            case UNIX_TIMESTAMP_MILLIS:
+                return safeParseDate(DateUtils::parseUnixTimestampMillisInstant).apply(value);
```

Comment:
Do we need to add this to marshallers as well in case there's a new model that contains a date object being added in the future?

Reply:
Marshalling already works, it's handled a bit different than unmarshalling.

https://github.com/aws/aws-sdk-java-v2/blob/master/core/protocols/aws-cbor-protocol/src/main/java/software/amazon/awssdk/protocols/cbor/internal/SdkCborGenerator.java#L44